File: Algorithms/122.best-time-to-buy-and-sell-stock-ii/best-time-to-buy-and-sell-stock-ii.go
func maxProfit(prices []int) int {
    maxProfit := 0  //最大利润
    for i := 0; i < len(prices)-1; i++ {
        if prices[i+1] > prices[i] {
            //如果后一笔比前一笔数额大，做减法，累积利润
            maxProfit = maxProfit + prices[i+1] - prices[i]
        }
    }
    return maxProfit
}

File: Algorithms/122.best-time-to-buy-and-sell-stock-ii/best-time-to-buy-and-sell-stock-ii.java
class Solution {
    public int maxProfit(int[] prices) {
        if (prices == null || prices.length <= 1) return 0;

        int profit = 0;
        for (int i = 1; i < prices.length; i++) {
            int diff = prices[i] - prices[i - 1];
            if (diff > 0) profit += diff;
        }

        return profit;
    }
}
